web/src/reducers/errors.js

```diff
@@ -0,0 +1,67 @@
+import {
+  ADMIN_CREATE_USER_SUCCESS,
+  ADMIN_CREATE_USER_ERROR,
+  ADMIN_EDIT_ACCOUNT_SUCCESS,
+  ADMIN_EDIT_ACCOUNT_ERROR,
+  ADMIN_EDIT_ME_SUCCESS,
+  ADMIN_EDIT_ME_ERROR
+} from '../actions/admin';
+
+import { t } from '../i18n';
+
+const getError = (rawMessage, fallback) => {
+  const [errorFamily, message] = rawMessage.split('.');
+  let error = t(`errors.${errorFamily}.${message}`, { defaultValue: false });
+  if (error === false) {
+    error = t(`errors.${message}`, { defaultValue: false });
+  }
+  return error === false ? fallback : error;
+};
+
+const initialState = {
+  addAccount: false,
+  editAccount: false,
+  editOwnAccount: false
+};
+
+const successActions = new Map([
+  [ADMIN_CREATE_USER_SUCCESS, 'addAccount'],
+  [ADMIN_EDIT_ACCOUNT_SUCCESS, 'editAccount'],
+  [ADMIN_EDIT_ME_SUCCESS, 'editOwnAccount']
+]);
+
+const errorActions = new Map([
+  [ADMIN_CREATE_USER_ERROR, ['addAccount', 'Unknown error creating account']],
+  [ADMIN_EDIT_ACCOUNT_ERROR, ['editAccount', 'Unknown error editing account']],
+  [ADMIN_EDIT_ME_ERROR, ['editOwnAccount', 'Unknown error editing account']]
+]);
```

Comment:
These two things map action symbols to state property names.  This ended up being a lot more compact than putting each of these into the switch below individually, and pretty readable I think?

Reply:
I really like this approach! Makes sense and I like how the content and logic are separated a bit.